toxcore/group.c

```diff
@@ -879,8 +879,8 @@ int group_peername(const Group_Chats *g_c, uint32_t groupnumber, int peernumber,
     }
 
     if (g->group[peernumber].nick_len == 0) {
-        memcpy(name, "Tox User", 8);
-        return 8;
+        memcpy(name, "", 0);
```

Comment:
https://github.com/TokTok/c-toxcore/issues/768#issuecomment-364663331

Reply:
Um, @tox-user, do you realize that memory copy of 0 length doesn't copy anything? i.e. you might as well just remove that function call as it does nothing.

Also, Tox defines "empty" data, i.e. [empty name](https://github.com/TokTok/c-toxcore/blob/d3b286cb434ed228e7b62cc70cb293e7a5554bfa/toxcore/tox.h#L1092-L1127), as arrays of 0 length, so just setting length to 0 should be enough, it shouldn't matter what the name array contains if the length is 0, it might as well just say "install gentoo" for all intents and purposes, because clients are supposed to read exactly 0 bytes of that name (i.e. none). Also, toxcore doesn't null-terminate data/strings in the API, but I guess you could.

Reply:
I'm with @nurupo here, neither memcpy nor `name[0]=0` are correct here.

Reply:
memcpy is wrong.
why not be nice and put a null byte there. it is utf-8 ?

```
    if (peer->nick_len == 0) {
        name[0] = '\0';
        return 0;
    }
```

this seems the only correct version to me.
why be lazy and leave the buffer as is?

Reply:
Just return 0 is enough. If you get 0, you can read exactly 0 bytes from the buffer, so it doesn't matter what's in the first position that you never read.

Reply:
still we could be nice about it

Reply:
@zoff99 that's not being nice, it's writing a byte into a buffer that contains no bytes. These values aren't C Strings that need to be `\0` terminated.